### PR TITLE
Support Fast CDR v2

### PR DESCRIPTION
**Pull request by MiguelCompany**
This makes the necessary changes to allow building with Fast CDR v2

Part of ros2/ros2#1530

It is built on top of #113

### CODE REVIEW

**Review by clalancette — CHANGES_REQUESTED** [review]
I think this one needs to be rebased on top #113 (at least, it seems to me that includes some of the same changes).

Once that is done I'll do another review.

File: rosidl_typesupport_fastrtps_c/package.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
   <build_export_depend>rosidl_typesupport_interface</build_export_depend>
 
   <!-- Needed for headers in this package, or the headers it generates, and also needs to be linked in downstream packages -->
-  <depend>fastcdr</depend>
+  <depend version_gte="2">fastcdr</depend>
```
Comment (clalancette) [inline rosidl_typesupport_fastrtps_c/package.xml:33]: Same comment as in https://github.com/ros2/rmw_fastrtps/pull/746#discussion_r1532601464

File: rosidl_typesupport_fastrtps_cpp/package.xml [hunk 2]
```diff
@@ -32,7 +32,7 @@
   <build_export_depend>rosidl_typesupport_interface</build_export_depend>
 
   <!-- Needed for headers in this package, or the headers it generates, and also needs to be linked in downstream packages -->
-  <depend>fastcdr</depend>
+  <depend version_gte="2">fastcdr</depend>
```
Comment (clalancette) [inline rosidl_typesupport_fastrtps_cpp/package.xml:35]: Same comment as in https://github.com/ros2/rmw_fastrtps/pull/746#discussion_r1532601464

**Review by clalancette — CHANGES_REQUESTED** [review]
A couple of small changes here.

File: rosidl_typesupport_fastrtps_c/package.xml (hunk 1 above)
Comment (MiguelCompany) [reply rosidl_typesupport_fastrtps_c/package.xml]: I did it, but didn't push: a725fbb

File: rosidl_typesupport_fastrtps_cpp/package.xml (hunk 2 above)
Comment (MiguelCompany) [reply rosidl_typesupport_fastrtps_cpp/package.xml]: I did it, but didn't push: a725fbb

**Review by clalancette — APPROVED** [review]

**Comment by MiguelCompany** [comment]
@clalancette The tests that were updated in #113 were not updated here. I've fixed that in 7fb44ae

**Pull request merged**